### PR TITLE
daemon: fix data race when accessing requestedRestart

### PR DESCRIPTION
**Pull request by zyga**
This was flagged by TIOBE as CID:NA-0e2ca076d1d5a16a07303509981ae85d

### CODE REVIEW

**Review by andrewphelpsj — APPROVED** [review]
Thanks @zyga and TIOBE!

**Review by MiguelPires — APPROVED** [review]
LGTM but there's another access to `d.requestedRestart` towards the end of the method that could use the same fix (line 587)

File: daemon/daemon.go [hunk 1]
```diff
@@ -495,12 +495,13 @@ func (d *Daemon) Stop(sigCh chan<- os.Signal) error {
...
 	restartSocket := d.restartSocket
 	rebootInfo := d.rebootInfo
+	requestedRestart := d.requestedRestart
 	d.mu.Unlock()
 
 	// before not accepting any new client connections we need to write the
 	// maintenance.json file for potential clients to see after the daemon stops
 	// responding so they can read it correctly and handle the maintenance
-	if err := d.updateMaintenanceFile(d.requestedRestart); err != nil {
+	if err := d.updateMaintenanceFile(requestedRestart); err != nil {
```
Comment (bboozzoo) [inline daemon/daemon.go:504]: as @MiguelPires pointed out there's one other place accessing d.requestRestart directly, and then one more accessing/modifying d.restartSocket

**Comment by zyga** [comment]
This needs some more work. I have alot bigger version but I didn't finish it.

**Comment by zyga** [comment]
Closing due to https://github.com/snapcore/snapd/pull/14153

**Pull request closed without merging**